spec/categorical_spec.rb

```diff
@@ -101,14 +101,27 @@
   end
   
   context "#cut" do
-    let(:dv) { Daru::Vector.new [1, 2, 5, 14] }
-    subject { dv.cut (0..20).step(5) }
+    context "close at right end" do
+      let(:dv) { Daru::Vector.new [1, 2, 5, 14] }
+      subject { dv.cut (0..20).step(5) }
+      
+      it { is_expected.to be_a Daru::Vector }
+      its(:type) { is_expected.to eq :category }
+      its(:size) { is_expected.to eq 4 }
+      its(:categories) { is_expected.to eq ['0-4', '5-9', '10-14', '15-19'] }
+      its(:to_a) { is_expected.to eq ['0-4', '0-4', '5-9', '10-14'] }
+    end
     
-    it { is_expected.to be_a Daru::Vector }
-    its(:type) { is_expected.to eq :category }
-    its(:size) { is_expected.to eq 4 }
-    its(:categories) { is_expected.to eq ['0-4', '5-9', '10-14', '15-19'] }
-    its(:to_a) { is_expected.to eq ['0-4', '0-4', '5-9', '10-14'] }
+    context "close at left end" do
+      let(:dv) { Daru::Vector.new [1, 2, 5, 14] }
+      subject { dv.cut (0..20).step(5), :left }
+      
+      it { is_expected.to be_a Daru::Vector }
+      its(:type) { is_expected.to eq :category }
+      its(:size) { is_expected.to eq 4 }
+      its(:categories) { is_expected.to eq ['1-5', '6-10', '11-15', '16-20'] }
+      its(:to_a) { is_expected.to eq ['1-5', '1-5', '1-5', '11-15'] }
+    end
```

Comment:
Its possible to convert a numerical variable into categorical variable. For example `heights` store measures of heights and we want to categorize them into categories low, medium and high:

```ruby
> heights = Daru::Vector.new [30, 35, 32, 50, 42, 51]
> height_cat = heights.cut [30, 40, 50, 60]
> height_cat.rename_categories '30-39' => 'low', '40-49' => 'medium', '50-59' => 'high'
> height_cat.frequencies
=> #<Daru::Vector(3)>
    low      3
 medium      1
   high      2
```
In regression soemtimes, categorizing the numerical variable gives us have a better fitting model.